kratos/processes/assign_scalar_input_to_entities_process.cpp

```diff
@@ -640,12 +636,12 @@ void AssignScalarInputToEntitiesProcess<TEntity, THistorical>::InternalAssignVal
 
     if(number_of_entities != 0) {
         const auto it_begin = r_entities_array.begin();
-
-        #pragma omp parallel for
-        for(int i = 0; i<number_of_entities; i++) {
-            auto it_entity = it_begin + i;
+        IndexPartition<std::size_t>(number_of_entities).for_each([&](std::size_t index){
+            auto it_entity = it_begin + index;
             SetValue(*it_entity, rVariable, Value);
-        }
+        });
+        //Or using Variable Utitlities
+        //VariableUtils().SetNonHistoricalVariable(rVariable, Value, r_entities_array);
```

Comment:
@philbucher can we do like this using VariableUtils

Reply:
yes
Then you also don't need the `if(number_of_entities != 0) {` check any more